src/theme/functions.php

```diff
@@ -82,4 +83,75 @@ function wpb_move_comment_field_to_bottom($fields) {
   }
   add_filter('comment_form_fields', 'wpb_move_comment_field_to_bottom');
 
+  function pwa_theme_get_manifest_path() {
+    return add_query_arg( PWA_THEME_MANIFEST_ARG, '1', site_url() );
+  }
+
+  function pwa_theme_register_query_vars( $vars ) {
+    // $vars[] = PWA_SW_QUERY_VAR;
+    $vars[] = PWA_THEME_MANIFEST_ARG;
+    return $vars;
+  }
+  add_filter( 'query_vars', 'pwa_theme_register_query_vars' );
+
+  function pwa_theme_render_custom_assets() {
+    global $wp_query;
+    if ( $wp_query->get( PWA_THEME_MANIFEST_ARG ) ) {
+      $theme_color = pwa_theme_get_theme_color();
+
+      $manifest = array(
+          'start_url'  => get_bloginfo( 'url' ),
+          'short_name' => get_bloginfo( 'name' ),
+          'name'       => get_bloginfo( 'name' ),
+          'display'    => 'standalone',
+          'background_color' => $theme_color,
+          'theme_color' => $theme_color,
+      );
+
+      $icon_48 = pwa_site_icon_url( 48 );
+
+      if ( $icon_48 ) {
+          $manifest[ 'icons' ] = array(
+              array(
+                  'src' => $icon_48,
+                  'sizes' => '48x48'
+              ),
+              array(
+                  'src' => pwa_site_icon_url( 192 ),
+                  'sizes' => '192x192'
+              ),
+              array(
+                  'src' => pwa_site_icon_url( 512 ),
+                  'sizes' => '512x512'
+              )
+          );
+      }
+
+      wp_send_json( $manifest );
```

Comment:
Do we need to make sure that no other `template_redirect` hooks are run afterwards with `die` or something?

Reply:
wp_send_json `die`s all by itself :)

Reply:
TIL! Thanks